pkg/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_crd_reconcile.go

```diff
@@ -181,3 +196,34 @@ func (c *crdReconcile) updateStoredVersion(ctx context.Context) error {
 
 	return nil
 }
+
+// desiredStoredVersions caculate the desired stored versions
+// If the "requiredStoredVersion" do not exist, do not update the current stored versions
+// If the "requiredStoredVersion" exist, delete the "removedCRDStoredVersion"
+func desiredStoredVersions(curStoredVersion []string, requiredStoredVersion, removedCRDStoredVersion string) []string {
+	// remove old versions from its status
+	newStoredVersions := make([]string, 0, len(curStoredVersion))
+	if len(removedCRDStoredVersion) == 0 {
+		return curStoredVersion
```

Comment:
Should this function also return curStoredVersion here when len(requiredStoredVersion) == 0?